### PR TITLE
first section for adding support for cfn_nag issue 73 - Added  and  a…

### PR DESCRIPTION
**Pull request by tmcelhattan**
first section for adding support for cfn_nag issue 73.

- Added attribute of `usage_plans` to API Gateway Stage resources for any API Gateway Stage that is referenced within a Usage Plan resource.
- Added attribute of `deployment_id` to API Gateway Stage resources for any API Gateway Stage that references an API Deployment resource as its `DeploymentId` property.

### CODE REVIEW

File: lib/cfn-model/model/api_gateway_stage.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+require_relative 'model_element'
+
+class AWS::ApiGateway::Stage  < ModelElement
+  attr_accessor :usage_plans, :deployment_id
```
Comment (erickascic) [inline lib/cfn-model/model/api_gateway_stage.rb]: usage_plan is singular?

Comment (erickascic) [reply lib/cfn-model/model/api_gateway_stage.rb]: can deployment_id be multiple?

File: lib/cfn-model/parser/api_gateway_stage_parser.rb [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    attach_usage_plans_to_api_stage(cfn_model: cfn_model, api_stage: api_stage)
+    attach_deployment_id_to_api_stage(cfn_model: cfn_model, api_stage: api_stage)
+    api_stage
+  end
+
+  private
+
+  def attach_usage_plans_to_api_stage(cfn_model:, api_stage:)
+    usage_plans = cfn_model.resources_by_type 'AWS::ApiGateway::UsagePlan'
+    usage_plans.select do |usage_plan|
```
Comment (erickascic) [inline lib/cfn-model/parser/api_gateway_stage_parser.rb]: these select blocks have a side effect.... if you are just iterating and building up an external data structure, i would use "each" instead.

File: lib/cfn-model/model/api_gateway_stage.rb (hunk 1 above)
Comment (tmcelhattan) [reply lib/cfn-model/model/api_gateway_stage.rb]: @erickascic  Yea usage_plan should be singular, will make that change. And the property for `DeploymentId` on the Stage resource only accepts a single value.

File: lib/cfn-model/parser/api_gateway_stage_parser.rb [hunk 3]
```diff
@@ -9,21 +9,21 @@ class ApiGatewayStageParser
...
 
-  def attach_usage_plans_to_api_stage(cfn_model:, api_stage:)
+  def attach_usage_plan_to_api_stage(cfn_model:, api_stage:)
     usage_plans = cfn_model.resources_by_type 'AWS::ApiGateway::UsagePlan'
-    usage_plans.select do |usage_plan|
+    usage_plans.each do |usage_plan|
       next if usage_plan.apiStages.nil?
-      usage_plan.apiStages.select do |up_api_stage|
+      usage_plan.apiStages.each do |up_api_stage|
         if up_api_stage['Stage'].is_a?(Hash) && up_api_stage['Stage'].key?('Ref')
```
Comment (erickascic) [inline lib/cfn-model/parser/api_gateway_stage_parser.rb]: anything in References that can hide this?  @pshelby recently added some methods there

Comment (pshelby) [reply lib/cfn-model/parser/api_gateway_stage_parser.rb]: @tmcelhattan [`References.resolve_resource_id()`](https://github.com/stelligent/cfn-model/blob/0a590d92131e96f53097abd9b4942f249337f41f/lib/cfn-model/model/references.rb#L32-L46) might be able to help here.  Lmk if you need more info.  [Example usage](https://github.com/stelligent/cfn-model/blob/0a590d92131e96f53097abd9b4942f249337f41f/lib/cfn-model/model/cfn_model.rb#L82)

**Pull request merged**